cou/steps/plan.py

```diff
@@ -596,7 +595,27 @@ async def _get_upgradable_hypervisors_machines(
     if cli_force:
         return nova_compute_machines
 
-    return await get_empty_hypervisors(nova_compute_units, analysis_result.model)
+    empty_hypervisors = await get_empty_hypervisors(nova_compute_units, analysis_result.model)
+
+    machine_to_unit_name = {
+        machine.machine_id: unit.name
+        for unit, machine in zip(nova_compute_units, nova_compute_machines)
+    }
+    empty_hypervisor_ids = {machine.machine_id for machine in empty_hypervisors}
+    all_hypervisor_ids = {machine.machine_id for machine in nova_compute_machines}
+    skipped_hypervisor_ids = all_hypervisor_ids - empty_hypervisor_ids
+
+    skipped_unit_names = ", ".join(
+        machine_to_unit_name[machine_id] for machine_id in sorted(skipped_hypervisor_ids)
+    )
+    logger.info("Skipped (non-empty) hypervisors: %s", skipped_unit_names)
+
+    selected_unit_names = ", ".join(
+        machine_to_unit_name[machine_id] for machine_id in sorted(empty_hypervisor_ids)
+    )
+    logger.info("Selected (empty) hypervisors: %s", selected_unit_names)
```

Comment:
Why this is not part of `get_empty_hypervisors` and like that, we would not need to have `machine_to_unit_name` or any issue what `get_empty_hypervisors` should return?

If we move it there, we should not use words like `Skipped` and `Selected`, but we can use `Found not-empty ...`. Also, the message with empty hypervisors should not be warning level?